cranelift/codegen/src/machinst/lower.rs

```diff
@@ -989,7 +989,9 @@ impl<'func, I: VCodeInst> LowerCtx for Lower<'func, I> {
 
     fn use_input_reg(&mut self, input: LowerInput) {
         debug!("use_input_reg: vreg {:?} is needed", input.reg);
-        self.vreg_needed[input.reg.get_index()] = true;
+        if input.reg.is_virtual() {
```

Comment:
Comment here for why the reg may not be virtual: we may directly return a machine register when we know that register holds the result of an opcode (by definition).